driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -126,7 +124,7 @@ public void run() {
             }
             reusedConnection = null;
             connections.add(connection);
-            connectionFutures.add(handleErrors(setKeyspaceAsync(connectionFuture, connection, keyspace), initExecutor));
```

Comment:
Just did a little testing and it seems that if you have an EventLoopGroup with a thread count < 2*hosts and you can get into a state where a thread is tied up for readTimeout and then it tries to borrow a connection and set a keyspace again...eventually it evens out.  If my ELG is at least 2*hosts threads the problem doesn't manifest.  As mentioned in previous comment we knew this was a problem with non-core connections (or if user changed the keyspace on us), its just more visible now since it affects core connections.